src/util/transifex-upload.js

```diff
@@ -0,0 +1,79 @@
+const apiBaseUrl = 'https://www.transifex.com/api/2/project';
+const resource = 'app-strings';
+const token = '1/9492ad42eff9cc8c2d5a141f8df3647570d7a641';
+
+export function doTransifexUpload(contents, project, success, fail) {
+  const url = `${apiBaseUrl}/${project}/resources/`;
+  const updateUrl = `${apiBaseUrl}/${project}/resource/${resource}/content/`;
+  const headers = {
+    Authorization: `Basic ${Buffer.from(`api:${token}`).toString('base64')}`,
+    'Content-Type': 'application/json',
+  };
+
+  const req = {
+    accept_translations: true,
+    i18n_type: 'KEYVALUEJSON',
+    name: resource,
+    slug: resource,
+    content: contents,
+  };
+
+  function handleResponse(text) {
+    let json;
+    try {
+      // transifex api returns Python dicts for some reason.
+      // Any way to get the api to return valid JSON?
+      json = JSON.parse(text);
```

Comment:
I'm confused, you can call JSON.parse but not .json()?

Reply:
Yes. The `parse` call is an attempt to see if the response returned is valid JSON. If it's not, I just call the success callback with the raw text response instead. This happens whether or not it was a successful operation, or if an error occurred.